### PR TITLE
quote handling

### PR DESCRIPTION
**Pull request by amiller**
- Good riddance to Gramine-RATLS. We're managing our own TLS now to have a trustless domain
- If the private key exists (sealed file), use it as the TLS key
- Otherwise, generate a fresh private key and save it
- Also generate the Certificate Signing Request. Output this to untrusted file
- If the certificate isn't available (untrusted file), wait for it

Later we could include some automation to run certbot appropriately to fulfill the CSR but that's not the business of the enclave

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -29,12 +39,32 @@ async fn main() {
...
+    let csr = create_csr(&tee_url, &pkey).unwrap();
+    let csr_pem_bytes = csr.to_pem().unwrap();
+    fs::write(CSR_PATH, csr_pem_bytes).await.expect("Failed to write csr to file");
+
+    let mut pk_bytes = pkey.public_key_to_pem().unwrap();
+    let mut csr_pem_bytes = csr.to_pem().unwrap();
+    pk_bytes.append(&mut csr_pem_bytes);
+    if let Ok(quote) = sgx_attest::sgx_attest(pk_bytes) {
+        // handle quote
+        log::info!("quote: {:?}", quote);
```
Comment (amiller) [inline src/main.rs]: prefer this to output to `untrustedhost/quote.dat`

**Review by amiller — COMMENTED** [review]
looks great, only request is putting the quote to a file

File: src/main.rs (hunk 1 above)
Comment (rpalakkal) [reply src/main.rs]: added

File: src/main.rs [hunk 2]
```diff
@@ -97,21 +97,30 @@ async fn main() {
...
+    {
+        log::info!("Waiting for cert ...");
+        while !Path::new(CERTIFICATE_PATH).exists() {
+            sleep(Duration::from_secs(1)).await;
+        }
+        log::info!("Cert found");
+        let cert = fs::read(CERTIFICATE_PATH).await.expect("cert not found");
+        let config =
+            RustlsConfig::from_pem(cert, pkey.private_key_to_pem_pkcs8().unwrap()).await.unwrap();
+        let addr = SocketAddr::from(([0, 0, 0, 0], 8001));
```
Comment (amiller) [inline src/main.rs]: ports are good as untrusted env parameters, give the host some fleixiblity

**Review by amiller — COMMENTED** [review]
Session handling looks fine

**Pull request closed without merging**